app/models/course.rb

```diff
@@ -751,6 +751,18 @@ def remove_carat_from_error_messages
     end
   end
 
+  def select_modern_language_subjects
+    subjects.select { |subject| subject.type == "ModernLanguagesSubject" }
```

Comment:
```suggestion
subjects.where(type: "ModernLanguagesSubject") so it's a little more performant 
```

Reply:
I would also just rename this to `modern_language_subjects`

Reply:
I've renamed the method. 

Using `.where` seems to produce the same result as `select` but i'm getting a weird error. Do you know why? 

```
undefined method `push' for #<ActiveRecord::AssociationRelation [#<ModernLanguagesSubject id: 34, type: "ModernLanguagesSubject", subject_code: "15", subject_name: "French">, #<ModernLanguagesSubject id: 36, type: "ModernLanguagesSubject", subject_code: "17", subject_name: "German">]>
```

Reply:
https://github.com/DFE-Digital/publish-teacher-training/pull/3113/commits/348be7ef5523a62883968f611d968ba4761845e0